### DISCUSSION TITLE
Metadata step fails while compiling csharp projects with nuget references

### DISCUSSION THREAD

**Discussion by Stoffelche** [dotnet/docfx#8909] [Q&A]
I have a bunch of dotnet framework projects, for which I am maintaining a help site. 
It worked with version 2.61, but now I updated to 2.67.5 and the metatdata step fails, during compilation of projects, with nuget references.
It is unable to resolve types and namespaces of the nuget package class libraries. When I build the projects in VStudion there is not problem. I assume this is a bug, but before reporting it as such, I want to make sure it really is. Or is there some configuration setting, whidch tells docfx, to handle these nuget packages?

**Comment by yufeih** [comment]
Does setting `allowCompilationErrors` to `true` works? We've introduced strict compilation checks in newer versions which was ignored in previous versions.

```jsonc
{
  "metadata": {
    // ...
    "allowCompilationErrors": true
  }
}
```

**Reply by Stoffelche** [reply]
Thanks for your help. I can try it, but still, there should not be any errors when compiling. There must be a reason, that it does not recognize the references from nuget packages

**Reply by Stoffelche** [reply]
Also, it is unclear to me, where in the metadata section this statement should go. 
my metadata section is an array of src/dest objects, thus it does not start with a { but with [

**Reply by KalleOlaviNiemitalo** [reply]
According to [class MetadataJsonItemConfig](https://github.com/dotnet/docfx/blob/b03e06e6631c8dacfe8df1780e3a3b9081619713/src/Microsoft.DocAsCode.Dotnet/MetadataJsonConfig.cs#L34-L88), "allowCompilationErrors" should be a property of the same JSON object as "src" and "dest".

**Reply by KalleOlaviNiemitalo** [reply]
Seems these properties would be worth documenting in [Properties for `metadata`](https://github.com/dotnet/docfx/blob/b03e06e6631c8dacfe8df1780e3a3b9081619713/docs/reference/docfx-json-reference.md#11-properties-for-metadata).  Or make `class MetadataJsonItemConfig` public and add a link from the config JSON documentation to the generated API documentation so that it is always up to date.

**Reply by Stoffelche** [reply]
Thanks for your help.
But anyway, the problem, that it does not recognize types from nuget packages is not fixed by this settings. 
What coud be the reason for this?
